deprecate hook configuration via marks/attributes (#9118)
pytest-dev/pytest: changelog/4562.deprecation.rst

> Deprecate configureing hook specs/impls using attribute/marks.

Reviewer: ```suggestion
Deprecate configuring hook specs/impls using attribute/marks.
```

> Instead ``use pytest.hookimpl`` and ``pytest.hookspec``.

Reviewer: ```suggestion

Instead use :ref:`pytest.hookimpl` and :ref:`pytest.hookspec`.

For more details, see the :ref:`docs <deprecate-hook-marks>`.
```

> Deprecate configuring hook specs/impls using attributes/marks.

Reviewer 2: I think it would be good to list the marks/attributes, so people reading can do a grep search.

Author: good point, will put it into the deprecation docs

Instead use :py:func:`pytest.hookimpl` and :py:func:`pytest.hookspec`.

> For more details, see the :ref:`docs <configuring-hook-specs-impls-using-markers>`.

Reviewer: ```suggestion
For more details, see the :ref:`docs <legacy-path-hooks-deprecated>`.
```
